corfudb-tools/src/main/java/org/corfudb/browser/CorfuBrowserEditorCommands.java

```diff
@@ -173,4 +173,6 @@ CorfuDynamicRecord editRecord(String namespace, String tableName,
      * @return table names with given 'streamTag'
      */
     List<CorfuStoreMetadata.TableName> listTablesForTag(@Nonnull String streamTag);
+
+    void resetTrimmedTable(String namespace, String table);
```

Comment:
Do we want to expose this API?

Since the start and finish cp records are appended bypassing the objectView causing it to not capture conflicts. 
If the resetTrimmedTable is executed in known circumstances, when we know compactor will not be executed, there wouldn't be any problem. But exposing the API might get us to that scenario right?

Reply:
Actually by design it is allowed to have multiple checkpointer running at the same time. Every checkpointer has unique id and if you check `scanCheckpointStream`, it checks the checkpoint id to only read the checkpoint entries generated by the same checkpointer. 

The second point is valid though. Given that we no longer need to call this from corfu browser tool, let me remove it.

Reply:
Sorry I wasn't clear, not capturing the conflict is not the actual problem. But I meant more like the same stream might end up having 2 cp entries and say this cp entry is the latest, the actual cp entry done by the compactor is not used leading to problems.  -> I mean all these in case this method is invoked without knowledge of the actual use case intended.

Reply:
I will just add that there are plenty of existing ways to cause data loss due to incorrect usage so adding a way to reset a trimmed stream should not be the cause of safety concerns. Maybe we can consider failing this command if the table isn't already trimmed by querying the stream address space first?

Reply:
> Sorry I wasn't clear, not capturing the conflict is not the actual problem. Since the checkpointers aren’t aware of each other, this reset can cause trouble when this commits after the actual cp by the compactor (in case it’s parallelly running for the same table) -> I mean all these in case this method is invoked without knowledge of the actual use case intended.

I don't get the question. Are you saying we should stop any clients from writing to the table?